codepropertygraph/src/main/scala/io/shiftleft/utils/TimeMetric.scala

```diff
@@ -0,0 +1,349 @@
+package io.shiftleft.utils
+
+import com.sun.management.OperatingSystemMXBean
+import org.slf4j.{Logger, LoggerFactory}
+
+import java.io.PrintWriter
+import java.lang.management.ManagementFactory
+import java.util
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.{Calendar, Date}
+import scala.collection.mutable
+case class TimeMetricRecordConfig(resultFile: String = "cpuandmemoryusage.csv", recordFreq: Int = 60000)
+case class MaxPrintBuffers(MAX_FULLNAME_BUFFER: Int = 90, MAX_NAME_BUFFER: Int = 55, MAX_DONE_TIME_BUFFER: Int = 40)
+object TimeMetric {
+  val cal = Calendar.getInstance()
+  // Stage wise performance will be only recorded if parallel recording is started through initialize()
+  val stagePerformance: mutable.Map[String, NumberProcessor] = mutable.Map()
+  private val start                                          = cal.getTime
+  private var subStageStartTime                              = start
+  private var newTime                                        = start
+  private var stageStartTime                                 = start
+  private val ACROSS_PROCESS_STAGE                           = "<ACROSS_ALL>"
+  private val STAGE_NOT_SET                                  = "<not set>"
+  private val STARTED                                        = "Started"
+  private val DONE                                           = "Done"
+  private var lastStageName                                  = STAGE_NOT_SET
+  private var lastSubStageName                               = STAGE_NOT_SET
+  private var additionalMetaData                             = STAGE_NOT_SET
+  private var alsoLogConsole                                 = false
+  private val logger: (String, Boolean) => Unit = (msg, subStage) => {
+    if (alsoLogConsole && (!subStage || !supressSubstages))
```

Comment:
Based on how this is used, should this boolean not be called `printToConsole`? Or something

Reply:
done